environments/pettingzoo_atari_adapter.py

```diff
@@ -0,0 +1,323 @@
+# Copyright 2022 AI Redefined Inc. <dev+cogment@ai-r.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Tuple
+
+import cogment
+import gymnasium as gymna
+import numpy as np
+import supersuit as ss
+from cogment.environment import EnvironmentSession
+
+from cogment_verse.constants import PLAYER_ACTOR_CLASS, WEB_ACTOR_NAME
+from cogment_verse.specs import EnvironmentSpecs
+from cogment_verse.specs.ndarray_serialization import deserialize_ndarray
+from cogment_verse.specs.ndarray_serialization import SerializationFormat
+
+from cogment_verse.utils import import_class
+
+log = logging.getLogger(__name__)
+
+
+def get_pz_player(observation: np.ndarray, actor_names: list) -> Tuple[str, int]:
+    """Get name and index for the petting zoo player. Note that it works specifically to Atari game"""
+    num_agents = len(actor_names)
+    indicators = observation[0, 0, -num_agents:]
+    idx = int(np.where(indicators)[0])
+    current_agent_name = actor_names[idx]
+
+    return current_agent_name, idx
+
+
+def get_rl_agent(current_pz_agent_name: str, actor_names: list) -> Tuple[str, int]:
+    """Get index and name for reinforcement leanring"""
+    if len(actor_names) == 1:
+        return (actor_names[0], 0)
+    idx = [count for (count, agent_name) in enumerate(actor_names) if agent_name == current_pz_agent_name][0]
+    return (actor_names[idx], idx)
+
+
+def atari_env_wrapper(env: gymna.Env) -> gymna.Env:
+    """Wrapper for atari env"""
+    env = ss.max_observation_v0(env, 2)
+    env = ss.frame_skip_v0(env, 4)
+    env = ss.clip_reward_v0(env, lower_bound=-1, upper_bound=1)
+    env = ss.color_reduction_v0(env, mode="B")
+    env = ss.resize_v1(env, x_size=84, y_size=84)
+    env = ss.frame_stack_v1(env, 4)
+    env = ss.agent_indicator_v0(env, type_only=False)
+
+    return env
+
+
+class PzEnvType(Enum):
+    CLASSIC = "classic"
+    ATARI = "atari"
+
+
+class Environment(ABC):
```

Comment:
Should be merged with the petting zoo adapter

Reply:
Done

Reply:
You forgot to remove this one

Reply:
I did. It was removed